app/components/post_body_additional_content/post_body_additional_content.js

```diff
@@ -124,11 +125,16 @@ export default class PostBodyAdditionalContent extends PureComponent {
                 imageUrl = `https://i.ytimg.com/vi/${videoId}/hqdefault.jpg`;
                 ImageCacheManager.cache(null, `https://i.ytimg.com/vi/${videoId}/default.jpg`, () => true);
             } else {
-                const {data} = await this.props.actions.getRedirectLocation(link);
-
                 let shortenedLink;
-                if (data && data.location) {
-                    shortenedLink = data.location;
+
+                if (expandedLink) {
+                    shortenedLink = expandedLink;
+                } else {
+                    const {data} = await this.props.actions.getRedirectLocation(link);
```

Comment:
Instead of awaiting this, it probably makes more sense to have it call the action and then return. Then, we can call this again when the `expandedLink` prop changes so that it can do the caching that happens after this. It would look something like
```javascript
componentDidUpdate(prevProps) { // This will replace the old componentWillReceiveProps since that call is deprecated anyway
    const link = this.props.expandedLink || this.props.link;
    const prevLink = prevProps.expandedLink || prevProps.link;
    if (link !== prevLink) {
        this.load(link);
    }
}

load = (link) => {
    let imageUrl;
    if (isImageLink(link)) {
        // TODO cache the image
    } else if (isYouTubeLink(link)) {
        // TODO cache the image and thumbnail
    } else if (!this.props.expandedLink) {
        this.props.actions.getRedirectLocation(link);
    }
}
```

Reply:
That's true, I'll change it to this structure, I forgot the component updates when the state updates. 

Also, this would allow to expand links that have been shortened multiple times.